app/src/main/java/com/humberto/tasky/agenda/presentation/AgendaViewModel.kt

```diff
@@ -60,13 +76,43 @@ class AgendaViewModel @Inject constructor(
         }
     }
 
+    private fun getAgendaForDate(selectedDate: LocalDate) {
+        viewModelScope.launch(Dispatchers.IO) {
+            _agendaState.update { state ->
+                state.copy(isLoadingAgendaItems = true)
+            }
+            localAgendaDataSource.getAgendaForDate(selectedDate)
+                .collect { agenda ->
+                    _agendaState.update { state ->
+                        state.copy(
+                            agendaItems = orderAgendaItems(agenda),
+                            isLoadingAgendaItems = false
+                        )
+                    }
+                }
```

Comment:
There is no need to switch to `Dispatchers.IO` here. Your `localAgendaDataSource` should be handling that internally (and considering that you use Room, it already switches to the IO thread internally), thus being safe to call on the main thread.

Reply:
But if don't switch I get IllegalStateException: Cannot access database on the main thread since it may potentially lock the UI for a long period of time.

Reply:
Room is main-safe 🤔 When you mark your DAO functions as suspending, then you should have no problem calling them inside `viewModelScope.launch {}`

Can you share a bit more where you are getting this error?

Reply:
they were not marked as suspending 😅